plugins/Jobs.cpp

```diff
@@ -220,128 +238,192 @@ bool BedrockPlugin_Jobs::processCommand(SQLite& db, BedrockCommand& command) {
+            if (SContains(job, "unique")) {
+                SQResult result;
+                SINFO("Unique flag was passed, checking existing job with name " << job["name"]);
+                if (!db.read("SELECT jobID, data "
+                             "FROM jobs "
+                             "WHERE name=" + SQ(job["name"]) + ";",
+                             result)) {
+                    throw "502 Select failed";
+                }
 
-            // If we found a job, but the data was different, we'll need to update it.
-            if (!result.empty()) {
-                updateJobID = SToInt64(result[0][0]);
-            }
-        }
+                // If we got a result, and it's data is the same as passed, we won't change anything.
+                if (!result.empty() && result[0][1] == job["data"]) {
+                    SINFO("Job already existed with matching data, and unique flag was passed, reusing existing job "
+                          << result[0][0]);
 
-        // If no "firstRun" was provided, use right now
-        const string& safeFirstRun = request["firstRun"].empty() ? SCURRENT_TIMESTAMP() : SQ(request["firstRun"]);
+                    // If we are calling GetJob, return early, there are no more jobs to create.
+                    if(SIEquals(request.methodLine, "CreateJob")){
+                        content["jobID"] = result[0][0];
+                        return true;
+                    }
 
-        // If no data was provided, use an empty object
-        const string& safeData = request["data"].empty() ? SQ("{}") : SQ(request["data"]);
+                    // Append new jobID to list of created jobs.
+                    jobIDs.push_back(result[0][0]);
+                    continue;
+                }
 
-        // If a repeat is provided, validate it
-        if (request.isSet("repeat")) {
-            if (request["repeat"].empty()) {
-                SWARN("Repeat is set in CreateJob, but is set to the empty string. Job Name: "
-                      << request["name"] << ", removing attribute.");
-                request.erase("repeat");
-            } else if (!_validateRepeat(request["repeat"])) {
-                throw "402 Malformed repeat";
+                // If we found a job, but the data was different, we'll need to update it.
+                if (!result.empty()) {
+                    updateJobID = SToInt64(result[0][0]);
+                }
             }
-        }
 
+            // If no "firstRun" was provided, use right now
+            const string& safeFirstRun = !SContains(job, "firstRun") || job["firstRun"].empty() ? SCURRENT_TIMESTAMP() : SQ(job["firstRun"]);
 
-        // If no priority set, set it
-        int64_t priority = request.isSet("priority") ? request.calc("priority") : JOBS_DEFAULT_PRIORITY;
+            // If no data was provided, use an empty object
+            const string& safeData = !SContains(job, "data") || job["data"].empty() ? SQ("{}") : SQ(job["data"]);
```

Comment:
When we use `CreateJob`, we are sure that all parameters are being sent by PHP, but I cannot assure the same is happening when we receive a JSON Array with multiples jobs, so if I try accessing an element that doesn't exist in the job map, I think it will be returning an `out_of_range ` exception.